logfire/_internal/cli.py

```diff
@@ -299,42 +302,41 @@ def _write_credentials(project_info: dict[str, Any], data_dir: Path, logfire_api
 def parse_create_new_project(args: argparse.Namespace) -> None:
     """Create a new project."""
     data_dir = Path(args.data_dir)
-    logfire_url = args.logfire_url
-    if logfire_url is None:  # pragma: no cover
-        _, logfire_url = LogfireCredentials._get_user_token_data()  # type: ignore
+    logfire_url: str | None = args.logfire_url
+    user_token = default_token_collection().get_token(logfire_url)
+    client = LogfireClient(user_token)
```

Comment:
Can also be added on the namespace to avoid repetition:

```python
namespace.get_client = lambda: LogfireClient(default_token_collection().get_token(namespace.logfire_url))
```